api/valor_api/backend/core/label.py

```diff
@@ -26,71 +26,95 @@ def validate_matching_label_keys(
     ----------
     db : Session
         The database Session to query against.
-    prediction_filter : schemas.Filter
-        The filter to be used to query predictions.
-    groundtruth_filter : schemas.Filter
-        The filter to be used to query groundtruths.
+    dataset_names : list[str]
+        The list of required datasets by name.
+    model_name : str
+        The required model by name.
     label_map: LabelMapType, optional
         Optional mapping of individual labels to a grouper label. Useful when you need to evaluate performance using labels that differ across datasets and models.
 
-
     Raises
     -------
     ValueError
         If the distinct ground truth label keys don't match the distinct prediction label keys for any datum.
     """
 
-    gts = generate_select(
```

Comment:
we use `generate_select` statements like this one in a few other places; could those be leaking data, too?

Reply:
Im reverting to using filters per your previous comment. These should not leak as it is now adding datasets, model, label_mapping and task_type to the select generation.

These were missing previously.

Reply:
great, thanks!